### PR TITLE
ci: Use black as the formatter

### PR DESCRIPTION
**Pull request by KShivendu**
## Summary of change

Use black as the formatter

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# run this command: git config --global blame.ignoreRevsFile .git-blame-ignore-revs
```
Comment (rishabhpoddar) [inline .git-blame-ignore-revs:1]: what is this file for?

File: dev-requirements.txt [hunk 2]
```diff
@@ -2,12 +2,12 @@ anyio==3.5.0
...
 astroid==2.9.3
 attrs==21.4.0
-autopep8==1.5.6
+black==22.3.0
 certifi==2021.10.8
 cffi==1.15.0
 chardet==4.0.0
 charset-normalizer==2.0.12
-click==7.1.2
+click>=8.0
```
Comment (rishabhpoddar) [inline dev-requirements.txt:10]: exact version here

File: setup.py [hunk 3]
```diff
@@ -13,24 +13,30 @@
...
-        'django',
-        'django-stubs==1.9.0',
-        'uvicorn==0.13.4',
-        'python-dotenv==0.19.2',
-    ]),
+    "fastapi": (
+        [
+            "respx==0.16.3",
+            "Fastapi",
+            "uvicorn==0.13.4",
```
Comment (rishabhpoddar) [inline setup.py:20]: this needs to change too.

File: setup.py [hunk 4]
```diff
@@ -13,24 +13,30 @@
...
+            "Flask",
+            "python-dotenv==0.19.2",
+        ]
+    ),
+    "django": (
+        [
+            "django-cors-headers==3.11.0",
+            "django",
+            "django-stubs==1.9.0",
+            "uvicorn==0.13.4",
```
Comment (rishabhpoddar) [inline setup.py:36]: this needs to change too.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: examples/with-django/with-thirdpartyemailpassword/manage.py [hunk 5]
```diff
@@ -6,7 +6,7 @@
```
Comment (rishabhpoddar) [inline examples/with-django/with-thirdpartyemailpassword/manage.py:6]: also edit the changelog

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: dev-requirements.txt (hunk 2 above)
Comment (KShivendu) [reply dev-requirements.txt:10]: done.

File: setup.py (hunk 3 above)
Comment (KShivendu) [reply setup.py:20]: done.

File: setup.py (hunk 4 above)
Comment (KShivendu) [reply setup.py:36]: done.

File: .git-blame-ignore-revs (hunk 1 above)
Comment (KShivendu) [reply .git-blame-ignore-revs:1]: Removed.

It ignores the commit where I ran the black commit to preserve the git blame. 
It doesn't do anything unless you run the command `git config --global blame.ignoreRevsFile .git-blame-ignore-revs`.

File: examples/with-django/with-thirdpartyemailpassword/manage.py (hunk 5 above)
Comment (KShivendu) [reply examples/with-django/with-thirdpartyemailpassword/manage.py:6]: done.

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**